### ISSUE TITLE
TF 2.1 Automatic Mixed Precision Training Loss Not Going Down

### ISSUE THREAD

**Issue by coded5282** [tensorflow/tensorflow#40996]
I'm trying to use automatic mixed precision in TensorFlow 2.1.0, but my training loss has not been going down.

I'm using the follow code snippets:

tf.keras.mixed_precision.experimental.set_policy('mixed_float16')

optimizer = tf.keras.optimizers.Adam(learning_rate=lr)
optimizer = tf.keras.mixed_precision.experimental.LossScaleOptimizer(optimizer, "dynamic")

def _compute_apply_gradients(x):
        with tf.GradientTape() as tape:
            loss, metric_value = _compute_loss(x)
            loss = _optimizer.get_scaled_loss(loss)
        gradients = tape.gradient(loss, _model.trainable_variables)
        gradients = _optimizer.get_unscaled_gradients(gradients)
        _optimizer.apply_gradients(zip(gradients, _model.trainable_variables))
        return loss, metric_value

For calculating my loss, I'm doing the following:

outputs_logits, _ = _model(images, is_training=True)
loss = tf.nn.softmax_cross_entropy_with_logits(logits=outputs_logits, labels=labels)

I also added dtype='float32' to the final layer of my model, which is a U-Net.

When training, my loss starts from a really high value and then gradually decreases down to ~2.x, which is normal for regular training, and then just stops from there on - never going down. I have TensorFlow 2.1, Python 3.7.7, and a GPU with compute capability > 7.0.

Could this be an issue in TensorFlow? Has anyone else had this issue?

Or could it be an issue with the loss function? Thanks

**Comment by amahendrakar** [comment]
@coded5282,
In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here along with output log. 

Also, please add three back ticks (i.e. ```) before and after your code to preserve the indentation. Alternatively, you can attach the `.py` or notebook file you are running. Thanks!
